### DISCUSSION TITLE
AGGrid - Input Mask - Password

### DISCUSSION THREAD

**Discussion by bajirama** [zauberzeug/nicegui#2830] [Q&A]
### Question

Looking for options in AGGrid  for column to display the password  as "*****"  - any thoughts 
thanks for your help

**Comment by vickorian** [comment]
You can just obfuscate the column programmatically when you fill the grid. That seems the easiest thing to do. Were there any other things you were trying to do with the data that would make that not appropriate?

**Reply by bajirama** [reply]
Thanks for the quick response,  I have a dialog where I get the user and password information and populate the record on the main screen. I also have an edit button on the main page, where the user can select a row and edit it. If I obfuscate the column programmatically, then I will not be able to get the value when I try to edit it.

**Reply by vickorian** [reply]
so with AgGrid, there is a backend to the editor that generates a message when an edit is committed. I'm using it for a project now. AgGrid itself has an event "onCellEditingStarted", but i'm not sure how it's used. If you want to test it you could set up a test function to extract the args from it.

```py
def on_cell_editing_started(msg):
     print(msg.args)
```

Then in the aggrid just attach it to the call with .on

```py
grid = ui.aggrid(...).on("onCellEditingStarted", on_cell_editing_started)
```

to see what comes out. There should be a variable that lets you keep track of what was in the cell before the editing started and the column and row the data was taken from.

You can also change the cell type in the coldefs variable. But that would require you create a custom cell type because i don't' think AgGrid has a password celltype by default, but I know that those do exist in some other javascript based tables.

**Reply by vickorian** [reply]
@bajirama 

It's also possible to write custom javascript to handle hiding the values, but i'm not sure how to implement that exactly with respect to nicegui's aggrid. If anyone knows i'd be interested.
